src/actions/actions.ts

```diff
@@ -1702,27 +1702,55 @@ class CommandCmdA extends BaseCommand {
 
 function searchCurrentWord(position: Position, vimState: VimState, direction: SearchDirection, isExact: boolean) {
     const currentWord = TextEditor.getWord(position);
-    if (currentWord === undefined) {
+
+    // If the search is going left then use `getWordLeft()` on position to start
+    // at the beginning of the word. This ensures that any matches happen
+    // outside of the currently selected word.
+    const searchStartCursorPosition = direction === SearchDirection.Backward
+      ? vimState.cursorPosition.getWordLeft(true)
+      : vimState.cursorPosition;
+
+    return createSearchStateAndMoveToMatch(currentWord, vimState, direction, isExact, searchStartCursorPosition);
+}
+
+function searchCurrentSelection (vimState: VimState, direction: SearchDirection) {
+    const selection = TextEditor.getSelection();
+    const end = new Position(selection.end.line, selection.end.character + 1);
+    const currentSelection = TextEditor.getText(selection.with(selection.start, end));
+
+    // Go back to Normal mode, otherwise the selection grows to the next match.
+    vimState.currentMode = ModeName.Normal;
+
+    // If the search is going left then use `getLeft()` on the selection start.
+    // If going right then use `getRight()` on the selection end. This ensures
+    // that any matches happen outside of the currently selected word.
+    const searchStartCursorPosition = direction === SearchDirection.Backward
+      ? vimState.lastVisualSelectionStart.getLeft()
+      : vimState.lastVisualSelectionEnd.getRight();
+
+    return createSearchStateAndMoveToMatch(currentSelection, vimState, direction, false, searchStartCursorPosition);
+}
+
+function createSearchStateAndMoveToMatch(
```

Comment:
Ohh wait a sec. I mean like this:
```
function createSearchStateAndMoveToMatch(args: {
 needle: string | undefined,
 vimState: VimState,
 direction: SearchDirection,
 isExact: boolean,
 searchStartCursorPosition: Position,
}
```

IMO it's important to use an object here so we associate names with the arguments. Otherwise when we pass in the 5 different args to the function it's hard to remember what each one is.